compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -94,15 +94,10 @@ private BuildProject(ProjectEnvironmentBuilder environmentBuilder, Path projectP
         for (ModuleId moduleId : currentPackage().moduleIds()) {
             Module module = currentPackage().module(moduleId);
             Optional<Path> modulePath = modulePath(moduleId);
-            if (module.documentIds().contains(documentId)) {
+            if (module.documentIds().contains(documentId) || module.testDocumentIds().contains(documentId)) {
                 if (modulePath.isPresent()) {
                     return Optional.of(modulePath.get().resolve(module.document(documentId).name()));
                 }
-            } else if (module.testDocumentIds().contains(documentId)) {
-                if (modulePath.isPresent()) {
-                    return Optional.of(modulePath.get()
-                            .resolve(ProjectConstants.TEST_DIR_NAME).resolve(module.document(documentId).name()));
```

Comment:
The `ProjectConstants.TEST_DIR_NAME` is removed here since it adds a duplicated `tests` directory in the path generated. Here, the `module.document(documentId).name())` gives the bal file path along with the `tests` directory. 

There weren't other usages found for this method except for the one that is added in this PR.

Reply:
The duplication happens because we give the document name as `TEST_DIR_NAME + "/" + documentName` + <document-name>` to get the proper directory structure within the testable jar creation. Changing this is fine but I doubt it will work on Windows.

Reply:
IMO, its safe to split the test document name and resolve the path in an OS independent way.

Reply:
Done